samples/base/Session.cpp

```diff
@@ -308,20 +330,20 @@ namespace darwin {
 
         std::size_t packet_size;
 
-        if (hdr.certitude_size > 1) {
-            packet_size = sizeof(hdr) + sizeof(unsigned int) * (hdr.certitude_size - 1);
+        if (hdr->certitude_size > 1) {
+            packet_size = sizeof(darwin_filter_packet_t) + sizeof(unsigned int) * (hdr->certitude_size - 1);
         } else {
-            packet_size = sizeof(hdr);
+            packet_size = sizeof(darwin_filter_packet_t);
```

Comment:
we don't take into account the DEFAULT_CERTITUDE_LIST_SIZE
the code works because the value of DEFAULT_CERTITUDE_LIST_SIZE is 1, but we should take it into account when making the calculations

Reply:
Corrected and improved in b3c45cf38eb225abc923993b3a3eb036feff74b5